headers/spi_comm.h

```diff
@@ -0,0 +1,255 @@
+/**
+ * @file spi_comm.h
```

Comment:
General nit:

Separating the '.h' and '.c' files into 2 different folders would be considered unconventional.

Imagine a large scale project with a lot of different file levels. You would have a _loooot_ of different 'header' folders